### PR TITLE
fix(Core/Spells): Hand of/Divine Sacrifice should remove remove the Hungering Cold frozen effect.

### PR DESCRIPTION
**Pull request by ErnestoGonzalezC**
## Changes Proposed:
-  Hand of/Divine Sacrifice now remove the Hungering Cold frozen effect.

## Issues Addressed:
- Closes 

## Tests Performed:

- Tested on windows 10 and ubuntu 20.

## How to Test the Changes:

- Log in with a paladin and any other character.
- Make a group with the two characters.
- While any npc is hitting his other character cast with the paladin `` Hand of sacrifice`` or `` Divine sacrifice``.
- Apply `` Hungering cold`` on yourself and verify that the damage generated by ``Divine sacrifice`` or ``Hand of sacrifice`` eliminates the effect of `` Hungering Cold``

## Target Branch(es):
- [x] Master

 
## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.

### CODE REVIEW

File: src/server/game/Entities/Unit/Unit.cpp [hunk 1]
```diff
@@ -2073,6 +2073,17 @@ void Unit::CalcAbsorbResist(Unit* attacker, Unit* victim, SpellSchoolMask school
...
+            {
+				case HAND_OF_SACRIFICE:
+                case DIVINE_SACRIFICE:
+                {
+					if (caster->HasAura(HUNGERING_COLD))
+						caster->RemoveAura(HUNGERING_COLD);
+                    break;
+                }
+            }
+
```
Comment (FrancescoBorzi) [inline src/server/game/Entities/Unit/Unit.cpp:2086]:
```suggestion
			switch ((*itr)->GetId())
            {
			    case HAND_OF_SACRIFICE:
			    case DIVINE_SACRIFICE:
				    if (caster->HasAura(HUNGERING_COLD))
				    {
				        caster->RemoveAura(HUNGERING_COLD);
			        }
			        break;
                default:
                    break;
            }

```

File: src/server/game/Entities/Unit/Unit.h [hunk 2]
```diff
@@ -136,6 +136,10 @@ enum SpellFacingFlags
 #define BASE_MAXDAMAGE 2.0f
 #define BASE_ATTACK_TIME 2000
 
+constexpr uint32 HAND_OF_SACRIFICE = 6940;
+constexpr uint32 DIVINE_SACRIFICE  = 64205;
+constexpr uint32 HUNGERING_COLD    = 51209;
+
```
Comment (FrancescoBorzi) [inline src/server/game/Entities/Unit/Unit.h:142]:
```suggestion
constexpr uint32 HAND_OF_SACRIFICE = 6940;
constexpr uint32 DIVINE_SACRIFICE = 64205;
constexpr uint32 HUNGERING_COLD = 51209;

```

**Comment by Mesielecat** [comment]
Why use a hack for this when this can be do it in a script?

**Comment by ErnestoGonzalezC** [comment]
> Why use a hack for this when this can be do it in a script?

This way it is much easier.

**Comment by FrancescoBorzi** [comment]
I agree with @Mesielecat , scripting should be preferred to core patching. Can you please suggest an alternative fix?

**Pull request closed without merging**